utils/build-script-impl

```diff
@@ -2401,6 +2403,41 @@ for host in "${ALL_HOSTS[@]}"; do
                 ;;
                 esac
 
+                ;;
+            foundation_macros)
+                if [[ ${host} == "macosx"* ]]; then
+                    echo "Skipping Foundation Macros on OS X -- Foundation is part of the OS on this platform"
+                    continue
+                fi
+
+                if [[ "${SKIP_CLEAN_FOUNDATION}" == "0" ]]
+                then
+                  # The Swift project might have been changed, but CMake might
+                  # not be aware and will not rebuild.
+                  echo "Cleaning the Foundation Macros build directory"
+                  call rm -rf "${build_dir}"
+                fi
+
+                cmake_options=(
+                    ${cmake_options[@]}
+                    -DCMAKE_BUILD_TYPE:STRING=${FOUNDATION_BUILD_TYPE}
+                    -DCMAKE_C_COMPILER:PATH=${CLANG_BIN}/clang
+                    -DCMAKE_CXX_COMPILER:PATH=${CLANG_BIN}/clang++
+                    -DCMAKE_SWIFT_COMPILER:PATH=${SWIFTC_BIN}
+                    -DCMAKE_Swift_COMPILER:PATH=${SWIFTC_BIN}
```

Comment:
Why do we need both variants? `CMAKE_Swift_COMPILER` is the version that CMake officially supports and `CMAKE_SWIFT_COMPILER` was something that we used in Swift to do builds.

Reply:
This was just copied from the existing cmake invocation for the rest of Foundation, I can remove the `CMAKE_SWIFT_COMPILER` one if it's no longer needed